File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <algorithm>
#include <vector>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    
    vector<int> answer;
    
    for(vector<int> command : commands) {
        vector<int> tmp;
        for(int i = command[0] - 1 ; i < command[1] ; i++)
            tmp.push_back(array[i]);
        // 정렬
        sort(tmp.begin(), tmp.end());
        answer.push_back(tmp[command[2] - 1]);
    }

    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        List<Integer> list = Arrays.stream(array).boxed().collect(Collectors.toList());
        List<Integer> ret = new ArrayList<>();
        for(int[] c : commands){
            List<Integer> sub = new ArrayList<>(list.subList(c[0] - 1, c[1]));
            Collections.sort(sub, Comparator.naturalOrder());
            ret.add(sub.get(c[2] - 1));
        }
        
        return ret.stream().mapToInt(Integer::valueOf).toArray();
    }
}